### PR TITLE
Add permission checks to parser functions

### PR DESCRIPTION
**Pull request by marijnvanwezel**
This pull request:

* Adds `read` permission checks to the `#slottemplates` and `#slot` parser functions.
* Formats all files.
* Removed an erroneous comment in Lua.

The files that should be checked in particular are `src/ParserFunctions/SlotParserFunction.php` and `src/ParserFunctions/SlotTemplatesParserFunction.php`.

### CODE REVIEW

File: src/ParserFunctions/SlotTemplatesParserFunction.php [hunk 1]
```diff
@@ -4,65 +4,79 @@
...
+	 *
+	 * @param Parser $parser
+	 * @param string $slotName
+	 * @param string|null $pageName
+	 * @param string|null $arrayName
+	 * @param string|null $recursive
+	 * @return string
+	 * @throws MWException
+	 */
+	public function execute( Parser $parser, string $slotName, string $pageName = null, string $arrayName = null, string $recursive = null ): string {
```
Comment (Designburo) [inline src/ParserFunctions/SlotTemplatesParserFunction.php:33]: Exceeds 120 chars use 
```
public function execute(
		Parser $parser,
		string $slotName,
		string $pageName = null,
		string $arrayName = null,
		string $recursive = null
	) : string {
```

Comment (Designburo) [reply src/ParserFunctions/SlotTemplatesParserFunction.php:33]: Didn't see any weird things, except for some mw codestyling mentioned. Why not use the userCan function, as it does the check for you ?

Comment (marijnvanwezel) [reply src/ParserFunctions/SlotTemplatesParserFunction.php]: Thanks. The code now uses `userCan`.

**Review by Designburo — APPROVED** [review]

**Comment by simontaurus** [comment]
[Use 'userCan' instead of 'getPermissionErrors'](https://github.com/Open-CSP/WSSlots/pull/2/commits/6ba791804973306bff992edfedb1b888da8c7d4b) is missing here: https://github.com/Open-CSP/WSSlots/blob/master/src/API/ApiEditSlot.php#L34 and here: https://github.com/Open-CSP/WSSlots/blob/master/src/API/ApiReadSlot.php#L32, isn't it?

**Comment by marijnvanwezel** [comment]
> [Use 'userCan' instead of 'getPermissionErrors'](https://github.com/Open-CSP/WSSlots/pull/2/commits/6ba791804973306bff992edfedb1b888da8c7d4b) is missing here: https://github.com/Open-CSP/WSSlots/blob/master/src/API/ApiEditSlot.php#L34 and here: https://github.com/Open-CSP/WSSlots/blob/master/src/API/ApiReadSlot.php#L32, isn't it?

Those files are not edited in this branch.

**Pull request merged**